app/data/tasks/deduplicate_records.py

```diff
@@ -0,0 +1,185 @@
+    generate it
+    """
+    queryset = Record.objects.filter(
+        occurred_from__range=(
+            extent['start_time'], extent['end_time']
+        )
+    ).exclude(uuid__in=get_dedupe_ids())
+    ids = list(queryset.values_list(queryset.model._meta.pk.name, flat=True))
+    return (ids, queryset)
+
+
+def get_dedupe_ids(job=None):
+    """ Get the ids of records to dedupe
+    return a set of uuids to check for duplicates.
+    """
+    queryset = None
+    if job:
+        queryset = RecordDuplicate.objects.filter(job=job)
+    else:
+        queryset = RecordDuplicate.objects.all()
+    ids = list(queryset.values_list('record', flat=True))
+    return ids
+
+
+@shared_task
+def find_duplicate_records():
+    """ Find all duplicate records
+    Duplicate records are classified as two records that occured at the same time
+    and location
+
+    Heuristic:
+    Records that have times within 1 day of each other
+    Records that are located ~100m from each other
+
+    Multiple tasks running:
+    Each job will only consider records added after the last task started.
+    This should allow multiple tasks to run at the same time without identifying
+    duplicates more than once.
+
+    Job failure:
+    On job failure, the task should be retried after deleting the duplicates
+    associated with the task.  This is not automatic at this time.
+    A job may fail in a way that the job's failed state is not updated to be ERROR
+    if the celery worker itself goes down.  Otherwise, the task status should be
+    set properly
+    """
+    time_extent = get_time_extent()
+    ids, queryset = get_dedupe_set(time_extent)
+    # this can be adjusted depending on the memory required
+    paginator = Paginator(ids, 1000)
+    job = DedupeJob(status=DedupeJob.Status.STARTED)
+    job.save()
+    starttime = datetime.datetime.now()
+    duplicatecount = 0
+    try:
+        logger.info("Finding duplicates. 1000 records/batch, %d batches total" %
+                    (paginator.num_pages))
+        for page_num in paginator.page_range:
+            page = paginator.page(page_num)
+            records = queryset.in_bulk(page.object_list)
+            done = []
+            for record in records:
+                done.append(record)
+                duplicates = find_duplicates_for_record(record)
+                recordDuplicates = []
+                for duplicate in duplicates:
+                    if duplicate.uuid not in done:
+                        # handle this in the find_duplicates_for_record call
+                        # to reduce number of queries
+                        num_reversed = RecordDuplicate.objects.filter(
+                            record__uuid=duplicate.uuid,
+                            duplicate_record__uuid=record
+                        ).count()
+                        if num_reversed == 0:
+                            rec = Record.objects.get(uuid=record)
```

Comment:
`Queryset.in_bulk` returns a dict of id/Model pairs, so it should be possible to avoid this extra database hit by iterating over `for rec_id, record in records.iteritems():`